tests/lib/testers/rule-tester.js

```diff
@@ -122,6 +122,17 @@ describe("RuleTester", () => {
         }, /Bad var\..*==.*Bad error message/);
     });
 
+    it("should throw an error when the error message regex does not match", () => {
+        assert.throws(() => {
+            ruleTester.run("no-var", require("../../fixtures/testers/rule-tester/no-var"), {
+                valid: [],
+                invalid: [
+                    { code: "var foo = bar;", errors: [{ message: /Bad error message/ }] }
+                ]
+            });
+        }, /Expected 'Bad var.' to match \/Bad error message\//);
+    });
+
     it("should throw an error when the error is neither an object nor a string", () => {
```

Comment:
@alberto - I changed the description to "should throw an error when the error is not a supported type", let me know if that's okay.